src/appleseed/foundation/utility/autoreleaseptr.h

```diff
@@ -147,6 +147,15 @@ class auto_release_ptr
         return m_ptr;
     }
 
+    // Bool conversion of the wrapped pointer.
+#if (defined _MSC_VER && _MSC_VER >= 1800) || defined(__GNUC__)
```

Comment:
I think this is correct, but an even better solution is this:

    #if !defined(_MSC_VER) || _MSC_VER >= 1800